src/Symfony/Component/Workflow/Marking.php

```diff
@@ -27,23 +27,58 @@ class Marking
     public function __construct(array $representation = [])
     {
         foreach ($representation as $place => $nbToken) {
-            $this->mark($place);
+            $this->mark($place, $nbToken);
         }
     }
 
-    public function mark(string $place): void
+    /**
+     * @param int $nbToken
+     */
+    public function mark(string $place /* , int $nbToken = 1 */): void
     {
-        $this->places[$place] = 1;
+        $nbToken = 1 < \func_num_args() ? func_get_arg(1) : 1;
+
+        if ($nbToken < 1) {
+            throw new \InvalidArgumentException(sprintf('The number of tokens must be greater than 0, "%s" given.', $nbToken));
+        }
+        dump(isset($this->place[$place]));
+        if (!\array_key_exists($place, $this->places)) {
+            $this->places[$place] = 0;
+        }
+        $this->places[$place] += $nbToken;
     }
 
-    public function unmark(string $place): void
+    /**
+     * @param int $nbToken
+     */
+    public function unmark(string $place /* , int $nbToken = 1 */): void
     {
-        unset($this->places[$place]);
+        $nbToken = 1 < \func_num_args() ? func_get_arg(1) : 1;
+
+        if ($nbToken < 1) {
+            throw new \InvalidArgumentException(sprintf('The number of tokens must be greater than 0, "%s" given.', $nbToken));
+        }
+
+        if (!$this->has($place)) {
+            throw new \InvalidArgumentException(sprintf('The place "%s" is not marked.', $place));
+        }
+
+        $nbToken = $this->places[$place] - $nbToken;
```

Comment:
I would suggest to use separate variable to represent the number of removed tokens and the number of tokens left in the place. It would make the code easier to understand than re-purposing the same variable (and will be needed anyway if you want to include the number of removed tokens in the error message)